### DISCUSSION TITLE
Vector Graphics on FvwmButtons

### DISCUSSION THREAD

**Discussion by 07AyubHaris** [fvwmorg/fvwm3#855] [Ideas]
## Describe the new feature

* What is it which fvwm3 doesn't do at the moment, but should?

## How might fvwm3 need to change?

* Non-technical thoughts about the impact this will have in `fvwm3`, what
  might need to change, and whether this impacts backwards-compatability.

I doubt FVWM3 would need to change very much to accommodate for this feature as the code for vectors already exists, however just needs to be applied to FvwmButtons. This would probably be achieved with "Vector" being an option inside brackets simmilar to this:

*FvwmButtons: (Top, Left, Title (Left) “title goes here", / 
Vector 8 35x15@1 20x50@1 40x50@0 30x80@1 75x40@0 50x40@1 75x15@0 35x15@1, )

## Can this be achieved through scripting?

* Is a proof-of-concept available externally through scripting `fvwm3`?
* Does this functionality exist elsewhere in other WMs, if so, which?

I am not sure if this effect can ba achieved through scripting and i am also not aware of this feature being available in any other window managers.

**Comment by lgsobalvarro** [comment]
I don't think this is a good idea. The `vector` command allows you to make rather simple/primitive shapes. Why extend its use when one can use svg images instead? In fact I would suggest the contrary: deprecate completely `vector` in favor of the use of svg.

**Reply by 07AyubHaris** [reply]
You Make a good point about SVG vector image files. However i rather like how vectors use the highlight and shadow colours and would rather this stay a feature within FVWM rather than relying on pixmaps.
